### PR TITLE
Turn CSI provisioner worker threads and timeout configurable

### PR DESCRIPTION
**Pull request by mhmxs**
${title}

### CODE REVIEW

File: api/v1/storageoscluster_types.go [hunk 1]
```diff
@@ -195,22 +195,24 @@ type ContainerImages struct {
...
+	KubeletDir                   string          `json:"kubeletDir,omitempty"`
+	PluginDir                    string          `json:"pluginDir,omitempty"`
+	DeviceDir                    string          `json:"deviceDir,omitempty"`
+	RegistrationDir              string          `json:"registrationDir,omitempty"`
+	KubeletRegistrationPath      string          `json:"kubeletRegistrationPath,omitempty"`
+	DriverRegistrationMode       string          `json:"driverRegisterationMode,omitempty"`
+	DriverRequiresAttachment     string          `json:"driverRequiresAttachment,omitempty"`
+	DeploymentStrategy           string          `json:"deploymentStrategy,omitempty"`
+	ProvisionerWorkerCount       uint8           `json:"provisionerWorkerCount,omitempty"`
+	Timeout                      metav1.Duration `json:"timeout,omitempty"`
```
Comment (Mojachieee) [inline api/v1/storageoscluster_types.go:215]: We should name this  `ProvisionerTimeout` or something, there's a lot of configurable timeouts, so it's good to be specific.

Comment (mhmxs) [reply api/v1/storageoscluster_types.go:215]: This is timeout for all external helper, does it make sense to make configurable each of them separately?

Comment (Mojachieee) [reply api/v1/storageoscluster_types.go]: Yeah, it does make sense to have separate timeouts for each of them, as some containers can take much longer (the snapshother has a default timeout of 60s vs the attacher was has a default of 15s). 

It's also ok to only make the provisioner timeout configurable, as that's the one we're seeing issues with

File: controllers/storageoscluster/csi_operand.go [hunk 2]
```diff
@@ -106,6 +107,74 @@ func (c *CSIOperand) Delete(ctx context.Context, obj client.Object) (eventv1.Rec
...
+			timeout = c.Spec.CSI.ResizerTimeout.Duration
+		}
+
+		return []string{
+			fmt.Sprintf(`[{ "op": "add", "path": "/spec/template/spec/containers/2/args/-", "value": "--timeout=%s" }]`, timeout),
+		}
+	},
+	"snapshotter": func(c *storageoscomv1.StorageOSCluster) []string {
+		const (
+			defaultTimeout = 30 * time.Second
```
Comment (Mojachieee) [inline controllers/storageoscluster/csi_operand.go:164]: Can we make this timeout 60s? As that's the original default

File: controllers/storageoscluster/csi_operand.go [hunk 3]
```diff
@@ -106,6 +107,74 @@ func (c *CSIOperand) Delete(ctx context.Context, obj client.Object) (eventv1.Rec
 	return nil, b.Delete(ctx)
 }
 
+// This collection depends on the order of containers at CSI deployment manifest,
+// on case of update please update the manifest too.
+var csiContainerPatches = map[string]func(c *storageoscomv1.StorageOSCluster) []string{
+	"provisioner": func(c *storageoscomv1.StorageOSCluster) []string {
+		const (
+			defaultWorkerThreads uint8 = 32
+			defaultTimeout             = 60 * time.Second
```
Comment (Mojachieee) [inline controllers/storageoscluster/csi_operand.go:116]: Can we make the provisioner timeout 30s, we did some testing and that works best

**Review by Mojachieee — APPROVED** [review]

**Pull request merged**